components/bl-transitions-component/src/collapse-top.js

```diff
@@ -10,26 +10,34 @@ export function CollapseTop(props) {
   const { onMounted, onUnmounted, onEndAnimation, onStartAnimation } = eventHandlers;
 
   const [height, setHeight] = useState(0);
+  const [initHeight, setInitHeight] = useState('');
 
   const rootRef = useRef();
+  const [element, setElement] = useState({ current: null });
 
-  const setIsAuto = useResizeObserver(rootRef.current, 'height', height, setHeight);
-  const isTransition = useTransition(rootRef, display, duration, height, 'height', setIsAuto, onEndAnimation);
-  const isImagesLoaded = useImageLoad(rootRef, dynamicContent);
+  const isImagesLoaded = useImageLoad(element, dynamicContent);
+  const isTransition = useTransition(element, display, duration, initHeight, height, 'height', onEndAnimation);
+
+  useEffect(() => {
+    if (rootRef.current && !element.current) {
+      setElement({ current: rootRef.current.firstElementChild });
+      setInitHeight(rootRef.current.firstElementChild.style.height);
+    }
```

Comment:
For example

```suggestion
    const readyToInitialTransition = rootRef.current && !element.current // or what we do here, i don't clearly undestand

    if (readyToInitialTransition) {
      setElement({ current: rootRef.current.firstElementChild });
      setInitHeight(rootRef.current.firstElementChild.style.height);
    }
```

This little things makes code more readable